pkg/plugins/crypto/impl/crypto.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+// Package impl provides the default implementation of pkg/plugins/crypto.Crypto interface
+//
+// It uses github.com/tink/go crypto primitives
+package impl
+
+import (
+	"errors"
+
+	"github.com/google/tink/go/aead"
+	"github.com/google/tink/go/core/primitiveset"
+	"github.com/google/tink/go/keyset"
+	aeadsubtle "github.com/google/tink/go/subtle/aead"
+	"golang.org/x/crypto/chacha20poly1305"
+
+	"github.com/hyperledger/aries-framework-go/pkg/plugins/crypto"
+)
+
+// DefaultCrypto is the default Crypto implementation
+type DefaultCrypto struct {
+}
+
+// Assert that DefaultCrypto implements the AriesCrypto interface.
+var _ crypto.AriesCrypto = (*DefaultCrypto)(nil)
+
+// Encrypt will encrypt msg using the implementation's corresponding encryption key
+func (t *DefaultCrypto) Encrypt(msg, aad []byte, kh *keyset.Handle) ([]byte, []byte, error) {
+	a, err := aead.New(kh)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	ct, err := a.Encrypt(msg, aad)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	ps, err := kh.Primitives()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	ivSize := getNonceSize(ps)
+	cipherText := ct[len(ps.Primary.Prefix)+ivSize:]
+	nonce := ct[len(ps.Primary.Prefix) : ivSize-1]
+
+	return cipherText, nonce, nil
+}
+
+func getNonceSize(ps *primitiveset.PrimitiveSet) int {
+	ivSize := aeadsubtle.AESGCMIVSize
+	// TODO double check if the right primitive from PrimitiveSet is Primary, or do we have to search for something else
+	if _, ok := (ps.Primary.Primitive).(aeadsubtle.XChaCha20Poly1305); ok {
+		ivSize = chacha20poly1305.NonceSizeX
+	}
+
+	return ivSize
+}
+
+// Decrypt will decrypt cipher using the implementation's corresponding encryption key
+func (t *DefaultCrypto) Decrypt(cipher, nonce, aad []byte, kh *keyset.Handle) ([]byte, error) {
+	a, err := aead.New(kh)
+	if err != nil {
+		return nil, err
+	}
+
+	ps, err := kh.Primitives()
+	if err != nil {
+		return nil, err
+	}
+
+	ct := make([]byte, 0, len(ps.Primary.Prefix)+len(nonce))
+	ct = append(ct, ps.Primary.Prefix...)
+	ct = append(ct, nonce...)
+	ct = append(ct, cipher...)
+
+	pt, err := a.Decrypt(ct, aad)
+	if err != nil {
+		return nil, err
+	}
+
+	return pt, nil
+}
+
+// Sign will sign msg using the implementation's corresponding signing key
+func (t *DefaultCrypto) Sign(msg []byte, kh *keyset.Handle) ([]byte, error) {
+	return nil, errors.New("not implemented yet")
```

Comment:
is this blocked or just not done yet?

Reply:
it's not blocked.. it will be added when it's needed..